app/controllers/admins_controller.rb

```diff
@@ -0,0 +1,19 @@
+class AdminsController < ApplicationController
+  before_action :authenticate_admin!
+
+  def index
+    @bookings = Booking.includes(:cleaner, :customer, :city).all
```

Comment:
sir in view, i need booking.customer.phone_number then if there is no includes it will get data from db every time loop iterates. I think that is not good for db. In index page i displayed all the bookings with its customer and cleaner details.